cumulus/parachains/runtimes/assets/asset-hub-rococo/tests/tests.rs

```diff
@@ -1355,3 +1356,17 @@ fn change_xcm_bridge_hub_ethereum_base_fee_by_governance_works() {
 		},
 	)
 }
+
+#[test]
+fn location_conversion_works() {
+	let alice_on_bh = Location::new(
+		1,
+		[
+			Parachain(1111),
+			Junction::AccountId32 { network: None, id: AccountId::from(ALICE).into() },
+		],
+	);
+	assert_ok!(LocationToAccountHelper::<AccountId, LocationToAccountId>::convert_location(
+		alice_on_bh.into()
+	));
```

Comment:
This `location_conversion_works` test currently checks one single location conversion: that of an `AccountId32` on a sibling parachain.

The test should verify multiple location types conversions. At the very least to completely cover regression testing for the changes in this PR it should cover **all branches** of `DescribeFamily<DescribeAllTerminal>`.

That means: 3 valid branches in [DescribeFamiliy](https://github.com/paritytech/polkadot-sdk/blob/7ecf3f757a5d6f622309cea7f788e8a547a5dce8/polkadot/xcm/xcm-builder/src/location_conversion.rs#L126C4-L142C21) x 6 valid types of location terminals defined in [DescribeAllTerminal](https://github.com/paritytech/polkadot-sdk/blob/7ecf3f757a5d6f622309cea7f788e8a547a5dce8/polkadot/xcm/xcm-builder/src/location_conversion.rs#L114C2-L119C23) = 18 valid locations types to verify.

Also, instead of `assert_ok!()`, you should check `assert_eq!()` against the hardcoded converted account, so that we know in the future this conversion doesn't accidentally change (breaking existing derived accounts).

Same comment applies for all the other runtimes tests.

Reply:
Applied requested changes to a single runtime. 
PTAL.